### PR TITLE
CI: Add GitHub Actions workflow to build and upload hex file.

### PR DESCRIPTION
**Pull request by microbit-carlos**
This way it's much easier to test any recent changes in-between releases.

Example CI run from my fork: https://github.com/microbit-carlos/micropython/runs/6025984640?check_suite_focus=true

### CODE REVIEW

**Comment by microbit-carlos** [comment]
@dpgeorge does this look okay?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          yotta --version
+      - name: Set up the Yotta target from GitHub
+        run: |
+          yotta target bbc-microbit-classic-gcc-nosd@https://github.com/lancaster-university/yotta-target-bbc-microbit-classic-gcc-nosd
+          yotta up
+      - run: make all
+      - name: Upload hex file
+        uses: actions/upload-artifact@v1
+        with:
+          name: microbitv1-micropython-${{ github.sha }}.hex
```
Comment (dpgeorge) [inline .github/workflows/build.yml:43]: maybe add the date to this filename?

Comment (microbit-carlos) [reply .github/workflows/build.yml]: Good suggestion! Done in https://github.com/bbcmicrobit/micropython/pull/737/commits/95524a2a2af8a5061733f2b390e7a22590677232.

**Comment by microbit-carlos** [comment]
I think this is ready for merge, thanks Damien!

**Pull request merged**